packages/design-system/src/components/component-card.tsx

```diff
@@ -0,0 +1,91 @@
+import { forwardRef, type ElementRef, ComponentProps } from "react";
+import { textVariants } from "../";
+import { css, theme } from "../stitches.config";
+
+const cardStyle = css({
+  display: "flex",
+  flexDirection: "column",
+  alignItems: "center",
+  justifyContent: "center",
+  px: theme.spacing[3],
+  width: theme.spacing[19],
+  height: theme.spacing[19],
+  border: `1px solid`,
+  borderColor: theme.colors.borderMain,
+  borderRadius: theme.borderRadius[2],
+  userSelect: "none",
+  color: theme.colors.foregroundIconMain,
+  cursor: "grab",
+  background: theme.colors.backgroundPanel,
+  "&:hover, &[data-state=hover]": {
+    background: theme.colors.backgroundHover,
+  },
+  "&[data-state=disabled]": {
+    background: theme.colors.backgroundPanel,
+    color: theme.colors.foregroundDisabled,
+  },
+  "&:focus-visible, &[data-state=focus]": {
+    outline: `2px solid ${theme.colors.borderFocus}`,
+    outlineOffset: "-2px",
+  },
+  "& svg": {
+    flexGrow: 0,
+    marginTop: theme.spacing[7],
+    width: 22,
+    height: 22,
+  },
+});
+
+const textStyle = css({
+  display: "flex",
+  flexAlign: "center",
+  justifyContent: "center",
+  flexDirection: "column",
+  flexGrow: 0,
+  height: "2em",
+  width: "100%",
+});
+
+const wordStyle = css(textVariants.small, {
+  display: "flex",
+  flexGrow: 0,
+  textAlign: "center",
+  justifyContent: "center",
+  whiteSpace: "nowrap",
+  textOverflow: "ellipsis",
+  overflow: "hidden",
+  width: "100%",
+});
```

Comment:
We could use https://caniuse.com/css-line-clamp for multiline text overflow

Reply:
(Just in case you didn't consider it)

Reply:
I was looking for something like this, didn't know it exists, but we also have another problem: single-word name is supposed to be centered vertically while the 2-words name would be split in 2 lines, where each word could be truncated

If we go this route with line-clamp, it would def. not allow ellipsis for each word, but maybe that's not super important after all, idk, I am sure @taylornowotny didn't know this is possible either